modules/health_quest/app/services/health_quest/pgd_service.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+Faraday::Response.register_middleware health_quest_errors: HealthQuest::Middleware::Response::Errors
+Faraday::Middleware.register_middleware health_quest_logging: HealthQuest::Middleware::HealthQuestLogging
+
+module HealthQuest
+  class PGDService < HealthQuest::SessionService
+    def get_pgd_resource(type, id = nil, pagination_params = {})
+      res_id = id || '333'
+      fhir_data = { id: res_id.to_s, text: "this is the #{type}" }
+      {
+        data: deserialized_resource({ my_attr: 'none/fake', fhir_data: fhir_data }, type),
+        meta: pagination(pagination_params)
+      }
+    end
+
+    private
+
+    def deserialized_resource(json_hash, type = nil)
+      result = json_hash[:fhir_data]
+      result[:type] = type if type
+      return [] unless result
+
+      result.is_a?(Array) ? result.map { |el| OpenStruct.new(el) } : OpenStruct.new(result)
+    end
+
+    def pagination(pagination_params)
+      {
+        pagination: {
+          current_page: pagination_params[:page] || 0,
+          per_page: pagination_params[:per_page] || 0,
+          total_pages: 0,
+          total_entries: 0
+        }
+      }
+    end
+
+    def get_pgd_base_url(type, id = nil)
```

Comment:
how is this method used?  It's in private but not used in the class

Reply:
get_pgd_base_url() would be used subsequently when the service connects to another backend system. Since the service is nonfunctional but the front end person wants to use it just to verify the endpoint is there or dummy results are returned, then it isn't being used other than the spec tests check it but I will have to remove it from the PR. 

The way the feature toggle works is that in the policy method, if it's not enabled then when the endpoint gets accessed it will return not authorized. I am going to add the feature being checked for to be "show_healthcare_experience_questionnaire" as currently it is using "va_online_scheduling"

Reply:
I don't understand how the flipper interacts with the PGDService you're introducing

Reply:
This PR doesn't have an endpoint for anything PGD, I will have to add that in a subsequent PR. Currently there is no way that anything can interact with the PGD service because there is not yet an associated endpoint. The feature toggle using Flipper.enabled?(<feature name>) works with endpoints only by checking for authorization. So the flipper should be working with endpoints and not the internal service.

Reply:
So you just described my concerns with merging this PR.  It's okay to make this part of a bigger PR that introduces the endpoint that uses this information.  Or better yet, create the endpoint pr first but return something like `render json: {data: 'dummy response'}` and then do this PR and connect them up.  But these changes aren't being used by anything and so wouldn't be a "viable" product (in a MVP sense).  So this puts the code-base at risk if the endpoint is never created to use this or (more likely) when the endpoint is created this code dosen't fit the way it needs to be used and has to be changed anyway.

Reply:
ok, well I think we can just abandon the dummy data strategy for now. I will push up other changes

Reply:
I have worked out a method to build a proxy to imitate an API using monteblank, that looks promising so the dummy data should not be needed. I have pushed up the service code without it returning dummy data